x/finality/keeper/gov.go

```diff
@@ -113,9 +120,38 @@ func (k Keeper) HandleResumeFinalityProposal(ctx sdk.Context, fpPksHex []string,
 
 		distCache.ApplyActiveFinalityProviders(params.MaxActiveFinalityProviders)
 
+		// add active fps to the all active fps set if it does not exist
+		for pk, fp := range distCache.GetActiveFinalityProviderSet() {
+			allActiveFps[pk] = fp
+		}
+
 		// set the voting power distribution cache of the current height
 		k.SetVotingPowerDistCache(ctx, h, distCache)
 	}
 
+	// it is possible that some inactive fps become active after the proposal
+	// therefore, we need to ensure every active finality provider has signing info
+	for pk, dc := range allActiveFps {
```

Comment:
@gitferry I had modified the logic here to avoid error equal checking, but let me know your thoughts I can rollback the changes

Reply:
Looks good to me! Thank you for taking this and adding more tests